src/topology/topological_fiber_bundle.lean

```diff
@@ -0,0 +1,476 @@
+  inv_fun    := λp, ⟨p.1, Z.coord_change i (Z.index_at p.1) p.1 p.2⟩,
+  to_fun     := λp, ⟨p.1, Z.coord_change (Z.index_at p.1) i p.1 p.2⟩,
+  map_source := λp hp,
+    by simpa only [set.mem_preimage, and_true, set.mem_univ, set.prod_mk_mem_set_prod_eq] using hp,
+  map_target := λp hp,
+    by simpa only [set.mem_preimage, and_true, set.mem_univ, set.mem_prod] using hp,
+  left_inv   := begin
+    assume p hx,
+    rcases p with ⟨x, v⟩,
+    change x ∈ Z.base_set i at hx,
+    dsimp,
+    rw [Z.coord_change_comp, Z.coord_change_self],
+    { exact Z.mem_base_set_at _ },
+    { simp [hx] }
+  end,
+  right_inv := begin
+    assume p hx,
+    rcases p with ⟨x, v⟩,
+    simp only [prod_mk_mem_set_prod_eq, and_true, mem_univ] at hx,
+    rw [Z.coord_change_comp, Z.coord_change_self],
+    { exact hx },
+    { simp [hx] }
+  end }
+
+@[simp] lemma mem_local_triv'_source (i : ι) (p : Z.total_space) :
+  p ∈ (Z.local_triv' i).source ↔ p.1 ∈ Z.base_set i :=
+by refl
+
+@[simp] lemma mem_local_triv'_target (i : ι) (p : B × F) :
+  p ∈ (Z.local_triv' i).target ↔ p.1 ∈ Z.base_set i :=
+by { erw [mem_prod], simp }
+
+@[simp] lemma local_triv'_fst (i : ι) (p : Z.total_space) :
+  ((Z.local_triv' i).to_fun p).1 = p.1 := rfl
+
+@[simp] lemma local_triv'_inv_fst (i : ι) (p : B × F) :
+  ((Z.local_triv' i).inv_fun p).1 = p.1 := rfl
+
+/-- The composition of two local trivializations is the trivialization change Z.triv_change i j. -/
+lemma local_triv'_trans (i j : ι) :
+  (Z.local_triv' i).symm.trans (Z.local_triv' j) ≈ (Z.triv_change i j).to_local_equiv :=
+begin
+  split,
+  { ext x, erw [mem_prod], simp [local_equiv.trans_source] },
+  { assume p hx,
+    rcases p with ⟨x, v⟩,
+    simp only [triv_change, local_triv', local_equiv.symm, true_and, local_equiv.right_inv,
+               prod_mk_mem_set_prod_eq, local_equiv.trans_source, mem_inter_eq, and_true,
+               mem_univ, prod.mk.inj_iff, local_equiv.trans_apply, mem_preimage, proj,
+               local_equiv.left_inv] at hx ⊢,
+    simp [Z.coord_change_comp, hx] }
+end
+
+/-- Topological structure on the total space of a topological bundle created from core, designed so
+that all the local trivialization are continuous. -/
+instance to_topological_space : topological_space Z.total_space :=
+topological_space.generate_from $ ⋃ (i : ι) (s : set (B × F)) (s_open : is_open s),
+  {(Z.local_triv' i).source ∩ (Z.local_triv' i).to_fun ⁻¹' s }
+
+lemma open_source' (i : ι) : is_open (Z.local_triv' i).source :=
+begin
+  apply topological_space.generate_open.basic,
+  simp only [exists_prop, mem_Union, mem_singleton_iff],
+  refine ⟨i, set.prod (Z.base_set i) univ, is_open_prod (Z.is_open_base_set i) (is_open_univ), _⟩,
+  ext p,
+  simp only [set.mem_preimage, and_true, set.mem_inter_eq, set.mem_univ,
+             topological_fiber_bundle_core.local_triv'_fst, iff_self, set.mem_prod, and_self,
+             topological_fiber_bundle_core.mem_local_triv'_source]
```

Comment:
This is finishing, you could leave `simp`.

Reply:
But that might make it substantially slower...?